In [0]:
import numpy as np
import pandas as pd

In [0]:
# Reading in the data from text files
def read_data_from_txt(filename, label):
  data = {}
  data['sentence'] = []
  data['label'] = []
  with open(filename, 'r') as f:
    for line in f:
      data['sentence'].append(line)
      data['label'].append(label)

  data_df = pd.DataFrame(data)
  return data_df

In [14]:
biased_data = read_data_from_txt('biased.txt', 1)
biased_data.head()

sentence  label
0  David J. Hanson Ph.D., Professor Emeritus of S...      1
1  While he continually puts forth these findings...      1
2  According to the Iranian government, the shoot...      1
3  Games revolving around the kicking of a ball t...      1
4  Irgun, shorthand for Irgun Tsvai Leumi, also s...      1

In [15]:
biased_data.shape

(1843, 2)

In [12]:
neutral_hard_data = read_data_from_txt('neutral_cw-hard.txt', 0)
neutral_hard_data.head()

sentence  label
0  The success of BJJ versus the other martial ar...      0
1  In all the games, the player is put in command...      0
2  Potwari ( , also known as Pothohari or Pahari-...      0
3  Hekmatyar's group, who, in October 1975, under...      0
4  After the Battle of Nicopolis in 1396 and the ...      0

In [13]:
neutral_hard_data.shape

(3109, 2)

In [16]:
neutral_featured_data = read_data_from_txt('neutral_featured.txt', 0)
neutral_featured_data.head()

sentence  label
0  Astatine was first produced by bombarding bism...      0
1  The hatched young, called squabs, are strongly...      0
2  In spite of early successes against invading I...      0
3  The Trojans trailed 10–3 going into halftime, ...      0
4  These bats possess long muzzles and long, exte...      0

In [17]:
neutral_featured_data.shape

(5000, 2)

In [18]:
neutral_type_balanced_data = read_data_from_txt('neutral_type_balanced.txt', 0)
neutral_type_balanced_data.head()

sentence  label
0  Other functions of the haemolymph include woun...      0
1  It was the fourth best-selling game of 2008, s...      0
2  Sherwood Springer's "No Land of Nod", in the s...      0
3  The Red Sox, winners of five of the first sixt...      0
4  The final part addresses the actual rise of al...      0

In [19]:
neutral_type_balanced_data.shape

(1994, 2)

### Number of Samples:
1.   Biased Data: 1843
2.   Neutral CW-Hard: 3109
3.   Neutral Featured: 5000
4.   Neutral Type-balanced: 1994

